### PR TITLE
ref(quick-start): Tweak completion logic and update 'create another project' task

### PR DESCRIPTION
**Pull request by priscilawebdev**
more details in the comments :)

**Note**
everything is behind a feature flag and only available for members of the telemetry-experience team

### CODE REVIEW

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 1]
```diff
@@ -291,6 +293,21 @@ export function getOnboardingTasks({
...
+        if (hasQuickStartUpdatesFeature(organization)) {
+          if (!projects?.length || task.requisiteTasks.length > 0 || taskIsDone(task)) {
+            return null;
+          }
+          return (
+            <EventWaitingIndicator
+              text={t('Waiting for error')}
+              hasQuickStartUpdatesFeature
+            />
+          );
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/taskConfig.tsx:306]: his task is only considered complete when the user sends an error event. Therefore, we should utilize the signalizer in the same way we do when waiting for the first error.

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 2]
```diff
@@ -282,7 +282,9 @@ export function getOnboardingTasks({
     },
     {
       task: OnboardingTaskKey.SECOND_PLATFORM,
-      title: t('Create another project'),
+      title: hasQuickStartUpdatesFeature(organization)
+        ? t('Set up another project')
+        : t('Create another project'),
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/taskConfig.tsx:287]: updated the name here as the creation of another project is not enough for completion

File: static/app/components/sidebar/newOnboardingStatus.tsx [hunk 3]
```diff
@@ -98,6 +99,17 @@ export function NewOnboardingStatus({
...
+  useEffect(() => {
+    if (pendingCompletionSeen && isActive && !hasMarkedUnseenTasksAsComplete.current) {
+      markDoneTaskAsComplete();
+      hasMarkedUnseenTasksAsComplete.current = true;
+    }
+
+    if (!pendingCompletionSeen || !isActive) {
+      hasMarkedUnseenTasksAsComplete.current = false;
+    }
+  }, [isActive, pendingCompletionSeen, markDoneTaskAsComplete]);
```
Comment (priscilawebdev) [inline static/app/components/sidebar/newOnboardingStatus.tsx:111]: Users can keep the sidebar open while completing tasks. Therefore, we shouldn't mark unseen tasks only when the sidebar is opened; we should also do so while it is active and users have completed tasks.

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 4]
```diff
@@ -291,6 +293,21 @@ export function getOnboardingTasks({
       actionType: 'app',
       location: `/organizations/${organization.slug}/projects/new/`,
       display: true,
+      SupplementComponent: withApi(({task}: FirstEventWaiterProps) => {
```
Comment (evanpurkhiser) [inline static/app/components/onboardingWizard/taskConfig.tsx:296]: Can you use the `useApi` hook instead?

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 5]
```diff
@@ -291,6 +293,21 @@ export function getOnboardingTasks({
...
+          }
+          return (
+            <EventWaitingIndicator
+              text={t('Waiting for error')}
+              hasQuickStartUpdatesFeature
+            />
+          );
+        }
+
+        return null;
```
Comment (evanpurkhiser) [inline static/app/components/onboardingWizard/taskConfig.tsx:309]: nit: I would flip this logic so it's easier to read and needs less indenting
```suggestion
        if (!hasQuickStartUpdatesFeature(organization)) {
          return null;
        }
        if (!projects?.length || task.requisiteTasks.length > 0 || taskIsDone(task)) {
          return null;
        }
        return (
          <EventWaitingIndicator
            text={t('Waiting for error')}
            hasQuickStartUpdatesFeature
          />
        );
```

File: static/app/components/onboardingWizard/taskConfig.tsx (hunk 4 above)
Comment (priscilawebdev) [reply static/app/components/onboardingWizard/taskConfig.tsx:296]: I could refactor this in a follow-up. For now I did not want to change it but reuse what we have in the code

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**